### PR TITLE
feat(align,fista): reduce XLA compiles and stabilize memory; fix scan body

### PR DESCRIPTION
**Pull request by tristanmanchester**
Summary
This PR improves stability and performance of alignment and FISTA by:

- Replacing Python chunk loops with a single lax.scan plus dynamic_slice and masked
tails to keep a single compiled jaxpr per batch size/resolution.
- Avoiding jnp.pad with traced pad widths; uses end-aligned masking with static-length
scans instead.
- Ensuring scan bodies receive explicit xs (fixes None input to scan).
- Laying groundwork for lower-compilation FISTA power-method selection.

Changes

- align/pipeline.py
    - Align loss over views now uses lax.scan + dynamic_slice with masked tails.
    - Minor logging tidy-ups for step metrics.
- recon/fista_tv.py
    - Batched data-term uses lax.scan + dynamic_slice and a single vmap body.
    - Power-method mode can be chosen based on views_per_batch (stream vs batched).
- utils/memory.py
    - No changes in this reset (only earlier commits touched).

Why

- Reduces repeated XLA compilations caused by Python loops over variable-sized chunks.
- Stabilizes memory by avoiding pad with traced sizes and using masked tails instead.
- Keeps a single, smaller jaxpr for both reconstruction and alignment passes,
improving cache reuse and build times.

Reproduction
CPU-friendly smoke tests (from pixi shell):

- pixi run install-root
- pixi run python -m pytest -q tests/test_projector.py
- pixi run python -m pytest -q tests/test_fista_tv.py
- pixi run python -m pytest -q tests/test_align.py

CLI checks:

- pixi run simulate --help
- pixi run recon --data data/sim.nxs --algo fbp --views-per-batch auto --out runs/
fbp.nxs
- pixi run recon --data data/sim.nxs --algo fista-tv --views-per-batch 1 --out runs/
fista.nxs
- pixi run align --data data/sim.nxs --levels 2 1 --outer-iters 3 --out runs/align.nxs

Performance Notes

- Device: CPU or small GPU is fine for verification; benefits on GPU are primarily
fewer compiles and more stable memory footprint for batched paths.
- Shapes: Use tests’ defaults for quick checks.
- Expect smaller jaxprs and fewer recompilations after the first run when
views_per_batch is fixed.

Compatibility

- No API breaks; only internal implementation changes.
- AlignConfig and FISTA signatures unchanged.

### CODE REVIEW

**Comment by tristanmanchester** [comment]
@codex fix comments
